processors/tracecompletiontime-deriver/src/test/java/org/hawkular/apm/processor/tracecompletiontime/TestCommunicationDetailsCache.java

```diff
@@ -61,4 +61,20 @@ public void store(String tenantId, List<CommunicationDetails> details) {
         }
     }
 
+    @Override
+    public List<CommunicationDetails> getById(String tenantId, String id) {
```

Comment:
Is this being used by the tests? Or used at all? If this code is not being used, and there's no intention to use it, wouldn't be better to throw an `UnsupportedOperationException`?